pkgs/checks/lib/src/checks.dart

```diff
@@ -187,40 +187,204 @@ extension ContextExtension<T> on Subject<T> {
   Context<T> get context => _context;
 }
 
-/// The expectation and nesting context already applied to a [Subject].
+/// The context for a [Subject] that allows asserting expectations and creating
+/// nested subjects.
 ///
-/// This is the surface of interaction for expectation extension method
-/// implementations.
+/// A [Subject] is the target for checking expectations in a test.
+/// Every subject has a [Context] which holds the "actual" value, tracks how the
+/// value was obtained, and can check expectations about the value.
 ///
-/// The `expect` and `expectAsync` can test the value and optionally reject it.
-/// The `nest` and `nestAsync` can test the value, and also extract some other
-/// property from it for further checking.
+/// The user focused APIs called within tests are expectation extension methods
+/// written in extension `on Subject`, typically specialized to a specific
+/// generic.
+///
+/// Expectation extension methods will make a call to one of the APIs on the
+/// subject's [Context], and can perform one of two types of operations:
+///
+/// -   Expect something of the current value.
+/// -   Expect that a new subject can be extracted from the current value.
```

Comment:
Would it be worth showing an example usage of each of these here, just to drive the point home a bit?
```suggestion
/// -   Expect something of the current value (`checkThat(1).equals(1)`).
/// -   Expect that a new subject can be extracted from the current value (`checkThat([]).length`).
```
Or you could have direct links to some real ones here?

Reply:
> Or you could have direct links to some real ones here?

It would introduce circular imports. I guess that wouldn't be a big deal within `src`...

Reply:
I went ahead and added the links, but they don't work currently.

https://github.com/dart-lang/dartdoc/issues/3325